File: ios/NativeMessage/MessageModule.swift
import Foundation
import Foundation
import React

@objc(MessageModule)
class MessageModule: NSObject {

  @objc func sendMessage(_ message: String, callback: @escaping RCTResponseSenderBlock) {
    let response = "Received from iOS: \(message)"
    callback([response])
  }

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

File: android/app/src/main/java/com/task/MessageModule.kt
package com.task

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Callback

class MessageModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
    override fun getName(): String {
        return "MessageModule"
    }

    @ReactMethod
    fun sendMessage(message: String, callback: Callback) {
        val response = "Received from Android: $message"
        callback.invoke(response) 
    }
}
